### PR TITLE
Add appimage build resolves issue #698

### LINKED ISSUE #698: [Feature request] AppImage target
It would be helpful if you could include some scripts to enable deploying of AppImage bundles as well as DEB/RPM packages.

I already have a recipe for building AppImage for GrandOrgue in my forked OBS repository: https://build.opensuse.org/package/view_file/home:lbssousa:grandorgue/grandorgue/appimage.yml?expand=1 , but I'm still having problems with JACK support. Maybe I should build GrandOrgue AppImage without JACK support.

This Github action can be handful: https://github.com/marketplace/actions/build-appimage

### PR DESCRIPTION
**Pull request by larspalo**
Resolves #698

### CODE REVIEW

File: build-scripts/for-appimage/prepare-debian-ubuntu.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+sudo apt update
+
+sudo DEBIAN_FRONTEND=noninteractive apt-get install -y cmake g++ pkg-config rpm file \
+  libjack-dev libfftw3-dev zlib1g-dev libasound2-dev libwavpack-dev libudev-dev \
+  libwxgtk3.0-gtk3-dev docbook-xsl xsltproc gettext po4a imagemagick patchelf libgtk-3-dev librsvg2-dev
+
```
Comment (oleg68) [inline build-scripts/for-appimage/prepare-debian-ubuntu.sh:10]: installing `rpm`, `file` and `libjack-dev` seems is not necessary for appimage build

File: build-scripts/for-appimage/build-on-linux.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (larspalo) [reply build-scripts/for-appimage/build-on-linux.sh:1]: Yes, that could possibly be an option in the future.

File: build-scripts/for-appimage/prepare-debian-ubuntu.sh (hunk 1 above)
Comment (larspalo) [reply build-scripts/for-appimage/prepare-debian-ubuntu.sh:10]: Right, that should work if there's no other hidden bug. I'll test it.

File: build-scripts/for-appimage-x86_64/build-on-linux.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+if [[ -n "$3" ]]; then
+	SRC_DIR=$3
+else
+	SRC_DIR=$(readlink -f $(dirname $0)/../..)
+fi
+
+PARALLEL_PRMS="-j$(nproc)"
+
+mkdir -p build-for/appimage
```
Comment (oleg68) [inline build-scripts/for-appimage-x86_64/build-on-linux.sh:19]:
```suggestion
mkdir -p build-for/appimage-x86_64
```

**Comment by oleg68** [comment]
@larspalo Could you alsso add the build instruction to BUILD.md?

**Comment by oleg68** [comment]
I've tested appimage 3.3.1-0.6 on Fedora34 and OracleLinux 8.4. It works now.

Today later I'll test on other distributions (OpenSuse, Ubuntu, Debian).

**Comment by oleg68** [comment]
@rousseldenis this PR is already waiting for your approvement in several days.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**